app/apollo/resolvers/common.js

```diff
@@ -55,53 +55,76 @@ const validClusterAuth = async (me, queryName, context) => {
+var getAllowedResources = async(me, org_id, action, type, queryName, context, searchByTags=null, searchByUuid=null, searchQuery=null)=>{
   const { models } = context;
-  var channels = await models.Channel.find({ org_id });
-  return await filterChannelsToAllowed(me, org_id, action, field, channels, context);
-};
 
-var filterChannelsToAllowed = async(me, org_id, action, field, channels, context)=>{
-  const { models } = context;
-  var decisionInputs = _.map(channels, (channel)=>{
-    return {
-      type: field,
-      action,
-      uuid: channel.uuid,
-      name: channel.name,
-    };
-  });
-  var decisions = await models.User.isAuthorizedBatch(me, org_id, decisionInputs, context);
-  channels = _.filter(channels, (val, idx)=>{
-    return decisions[idx];
-  });
-  return channels;
-};
+  var resources;
 
-var getAllowedSubscriptions = async(me, org_id, action, field, context)=>{
-  const { models } = context;
-  var subscriptions = await models.Subscription.find({ org_id });
-  return await filterSubscriptionsToAllowed(me, org_id, action, field, subscriptions, context);
+  // Check for cached IAM decision, returns true if cache is empty or all is authorized
+  const allAllowed = await cacheAllAllowed(me, org_id, action, type, queryName, context);
+
+  // Create to find by resource type
+  const modelType = type.charAt(0).toUpperCase() + type.slice(1);
+
+  // Find by resource type
+  if (searchByTags) {
+    resources = await models[modelType].find({org_id, tags: {$all: searchByTags}});
+  }
+  else if (type === 'group') {
+    if (searchByUuid) {
+      resources = await models[modelType].find({org_id, uuid: {$in: searchByUuid}});
+    }
+    else {
+      resources = await models[modelType].find({org_id}).lean({virtuals: true});
+    }
+  }
+  else if (type === 'subscription') {
+    if (searchQuery) {
+      resources = await models[modelType].find(searchQuery).lean({virtuals: true});
+    }
+  }
+  else {
+    resources = await models[modelType].find({org_id});
+  }
+
+  // If cache exists and all is not authorized filter through fine grained authentication
+  if (!allAllowed) {
+    return await filterResourcesToAllowed(me, org_id, action, type, resources, context);
+  }
+
+  return resources;
 };
 
-var filterSubscriptionsToAllowed = async(me, org_id, action, field, subscriptions, context)=>{
+// return user permitted resources in an array
+var filterResourcesToAllowed = async(me, org_id, action, field, resources, context)=>{
   const { models } = context;
-  var decisionInputs = _.map(subscriptions, (subscription)=>{
-    return {
-      type: field,
-      action,
-      uuid: subscription.uuid,
-      name: subscription.name,
-    };
+  var decisionInputs = _.map(resources, (resource)=>{
+    if (field === 'cluster'){
+      return {
+        type: field,
+        action,
+        uuid: resource.cluster_id,
```

Comment:
code in cluster.js uses `cluster.id` not `cluster.cluster_id` -- why the difference?  Is one incorrect?

Reply:
Poor naming convention used throughout Razeedash-api. They are interchangeable. I changed all I was working with to cluster.cluster_id as that was the one used in schema and most widely used.